pkg/cmd/cluster/cluster_unlink.go

```diff
@@ -0,0 +1,57 @@
+package cluster
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/dcos/dcos-cli/api"
+	"github.com/dcos/dcos-cli/pkg/config"
+	"github.com/spf13/cobra"
+)
+
+// newCmdClusterLink links the attached cluster to another one.
+func newCmdClusterUnlink(ctx api.Context) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:  "unlink",
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			attachedCluster, err := ctx.Cluster()
+			if err != nil {
+				return err
+			}
+
+			var linkedCluster config.Cluster
+			for _, cluster := range ctx.Clusters() {
+				if args[0] == cluster.ID() ||
+					args[0] == cluster.URL() ||
```

Comment:
In the current CLI, `dcos cluster unlink` doesn't accept an URL as argument (only a cluster name or id), I think it should stay like that.

The cluster linker API accepts an ID, and it might not be a configured cluster on the client side, so in case there is no cluster locally for a given ID, it should still make the DELETE request.

Reply:
In that case, why do we accept a cluster name? The URL is a unique argument that can easily be used to find which cluster should be unlinked, as having it accepted by the command requiring only one line I don't see why it shouldn't be allowed.

Reply:
URLs are not identifiers for machines or clusters, but resources. They also specify how to access these resources.

A DC/OS cluster has an unbounded number of resources under it. Even though we agreed on a given resource to "refer" to the cluster (eg. `/`), there would still be an unbounded number of URLs to identify the cluster (eg. using different schemes, specifying the port or not). If we were to go that route, I'd argue that only the host should be used to identify the cluster, but even then, different hosts can point to the same machine.

On the other hand, a DC/OS cluster has a single ID and a single name. They are unambiguous and canonical ways to identify a cluster. This is what we use to refer to them in `dcos cluster attach`, `dcos cluster remove` and `dcos cluster rename`. If we use the same code here (with the config manager), this would also remove more than a line of code.